yarn-project/pxe/src/pxe_oracle_interface/pxe_oracle_interface.test.ts

```diff
@@ -563,6 +565,109 @@ describe('PXEOracleInterface', () => {
     });
   });
 
+  describe('getLogByTag', () => {
+    const tag = Fr.random();
+    let scopedLog: TxScopedL2Log;
+    let txEffect: TxEffect;
+
+    beforeEach(async () => {
+      scopedLog = await TxScopedL2Log.random();
+      txEffect = await TxEffect.random();
+      aztecNode.getLogsByTags.mockReset();
+      aztecNode.getTxEffect.mockReset();
+    });
+
+    it('returns null if no logs found for tag', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[]]);
+
+      const result = await pxeOracleInterface.getLogByTag(tag);
+      expect(result).toBeNull();
+    });
+
+    it('returns log data when single log found', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog]]);
+      aztecNode.getTxEffect.mockResolvedValue(randomInBlock(txEffect));
```

Comment:
```suggestion
      aztecNode.getTxEffect.mockResolvedValue((txHash) => 
        txHash == scopedLog.txHash ? randomInBlock(txEffect) : undefined
      );
```

Costs us very little in terms of readability and makes it clearer what's going on.